src/goto-programs/abstract-interpretation/interval_analysis.cpp

```diff
@@ -39,18 +39,81 @@ static inline void simplify_guard(expr2tc &expr, const interval_domaint &state)
   simplify(expr);
 }
 
+static void optimize_expression(expr2tc &expr, const interval_domaint &state)
+{
+  ///////////////////
+  // PRECONDITIONS //
+  ///////////////////
+  if(is_nil_expr(expr))
+    return;
+
+  // We can't simplify addr-of sub-expr
```

Comment:
Why not?

Reply:
I will add a comment. 

But `int x = 3; int *ptr = &x;` would become `int x = 3; int *ptr = &3;`

Reply:
Yeah, that wouldn't make sense. But `&*ptr` could be simplified. However, I'm not entirely clear in what way this overlaps with what the irep2 simplifier is already doing.

Reply:
Hm, that could be done with the value set analysis.

Reply:
I have updated the comment.